### DISCUSSION TITLE
[3.x] Static typing without Variant?

### DISCUSSION THREAD

**Discussion by IPcorps** [godotengine/godot-proposals#5355] [Scripting]
I am aware of the benefits of static typing and I like good style. And at the same time:

`
func _get(property: String) -> Variant:
....
`

as a result: **error: The identifier "Variant" isn't a valid type**
This is strange, given that it is ["The most important data type in Godot"](https://docs.godotengine.org/en/3.5/classes/class_variant.html#variant). When I read TypeScript and see **_any_**, I know what the function returns. When I read GDScript and don't see anything (because then it would be Variant), I think "Oops! I forgot to type here!😯" Why not add the ability to specify a Variant without an error for the code analyzer? After all, in fact, all that needs to be done is simply to remove the error when specifying it.

**Comment by mieldepoche** [comment]
This is fixed in 4.0, maybe its impossible to backport.

**Reply by Calinou** [reply]
Feel free to attempt backporting it to `3.x`, but it's not a matter of simply copying some code around. The GDScript implementation in `3.x` is entirely different from the one in `master`.
